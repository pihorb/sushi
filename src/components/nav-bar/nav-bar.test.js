import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import { Router } from 'react-router'
import { createMemoryHistory } from 'history'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import NavBar from './nav-bar'

afterEach(cleanup)

const renderWithRouter = (component) => {
  const history = createMemoryHistory()
  return <Router history={history}>{component}</Router>
}

it('home page', () => {
  const { container, getByTestId } = render(renderWithRouter(<NavBar />))
  const navbar = getByTestId('navbar')
  const link = getByTestId('home-link')

  expect(container.innerHTML).toMatch('Home')
  expect(navbar).toContainElement(link)
})

it('menu page', () => {
  const { container, getByTestId } = render(renderWithRouter(<NavBar />))
  const navbar = getByTestId('navbar')
  const link = getByTestId('menu-link')

  expect(navbar).toContainElement(link)

  fireEvent.click(link)
  expect(container.innerHTML).toMatch('Menu')
})

it('delivery page', () => {
  const { container, getByTestId } = render(renderWithRouter(<NavBar />))
  const navbar = getByTestId('navbar')
  const link = getByTestId('delivery-link')

  expect(navbar).toContainElement(link)

  fireEvent.click(link)
  expect(container.innerHTML).toMatch('Delivery')
})

it('contact page', () => {
  const { container, getByTestId } = render(renderWithRouter(<NavBar />))
  const navbar = getByTestId('navbar')
  const link = getByTestId('contacts-link')

  expect(navbar).toContainElement(link)

  fireEvent.click(link)
  expect(container.innerHTML).toMatch('Contacts')
})

it('check links amount', () => {
  const { getAllByRole } = render(renderWithRouter(<NavBar />))
  const navItems = getAllByRole('listitem').length

  expect(navItems).toBe(4)
})

it('toggle class properly', () => {
  const { getByTestId } = render(renderWithRouter(<NavBar />))
  const btn = getByTestId('toggle-img')
  const list = getByTestId('list')

  expect(list).not.toHaveClass('open')

  fireEvent.click(btn)
  expect(list).toHaveClass('open')
})

it('snapshot', () => {
  const tree = renderer.create(renderWithRouter(<NavBar />)).toJSON()
  expect(tree).toMatchSnapshot()
})
