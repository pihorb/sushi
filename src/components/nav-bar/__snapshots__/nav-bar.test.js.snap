// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="nav"
  data-testid="navbar"
>
  <img
    alt="logo"
    className="nav__img"
    src="logo-min.png"
  />
  <ul
    className="nav__list"
    data-testid="list"
  >
    <li
      className="nav__item"
      data-testid="home-link"
      onClick={[Function]}
    >
      <a
        className="nav__link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav__item"
      data-testid="menu-link"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        className="nav__link"
        href="/"
        onClick={[Function]}
      >
        Menu
      </a>
    </li>
    <li
      className="nav__item"
      data-testid="delivery-link"
      onClick={[Function]}
    >
      <a
        className="nav__link"
        href="/delivery"
        onClick={[Function]}
      >
        Delivery
      </a>
    </li>
    <li
      className="nav__item"
      data-testid="contacts-link"
      onClick={[Function]}
    >
      <a
        className="nav__link"
        href="/contacts"
        onClick={[Function]}
      >
        Contacts
      </a>
    </li>
  </ul>
  <img
    alt="logo"
    className="nav__img menu-img"
    data-testid="toggle-img"
    onClick={[Function]}
    src="hamburger.png"
  />
</div>
`;
